src/components/Grid/Grid.js

```diff
@@ -2,14 +2,35 @@ import React from 'react';
 import styled from 'styled-components';
 import GridCard from 'components/GridCard/GridCard';
 
-const GridComponent = ({className, data}) =>  {
+const GridComponent = ({className, data: { page, totalPages, parsedData = [], isLoading }, pagination, setPageNumber}) =>  {
+
+    const createPaginationNumeration = (page, totalPages) => {
```

Comment:
Also this two parts of the file are in my opinion something you can move onto its own component or function and here just do some thing like 
```suggestion
    const paginationControls = pagination && <Paginator />
```